File: streaming-service/src/main/java/com/musicapp/streamingservice/config/MinioConfig.java
package com.musicapp.streamingservice.config;

import io.minio.MinioClient;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class MinioConfig {

    @Bean
    MinioClient minio(MinioProps props) {
        return MinioClient.builder()
                .endpoint(props.url())
                .credentials(props.accessKey(), props.secretKey())
                .build();
    }
}

File: image-service/src/main/kotlin/com/musicapps/imageservice/config/MinioConfig.kt
package com.musicapps.imageservice.config

import io.minio.MinioClient
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(MinioProps::class)
class MinioConfig {
    @Bean
    fun minio(props: MinioProps): MinioClient {
        return MinioClient.builder()
            .endpoint(props.url)
            .credentials(props.accessKey, props.secretKey)
            .build()
    }
}